networks/p2p/peer.go

```diff
@@ -353,6 +353,9 @@ loop:
 		case errRW.err = <-p.disc:
 			reason = discReasonForError(errRW.err)
 			break loop
+		case <-p.closed:
```

Comment:
It is not directly related with this PR, but some code is duplicated for `BaseSever`. The duplicate codes may continue to cause similar problems to the current issue. For example, I think that updating connection metrics is also different from the `BaseServer`'s code.

Reply:
I agreed your opinion. code for multichannel is a super set of code for single channel. so it will be removed and unified.